src/fsharp/vs/service.fsi

```diff
@@ -385,21 +401,18 @@ type internal FSharpChecker =
     ///
     /// <param name="filename">The filename for the file, used to help caching of results.</param>
     /// <param name="source">The full source for the file.</param>
-    /// <param name="options">The options for the project or script, used to determine active --define conditionals and other options relevant to parsing.</param>
-    /// <param name="userOpName">An optional string used for tracing compiler operations associated with this request.</param>
-    member MatchBraces : filename : string * source: string * options: FSharpProjectOptions * ?userOpName: string -> Async<(range * range)[]>
+    /// <param name="options">Parsing options for the project or script.</param>
+    member MatchBraces: filename: string * source: string * options: FSharpParsingOptions -> Async<(range * range)[]>
 
     /// <summary>
     /// <para>Parse a source code file, returning a handle that can be used for obtaining navigation bar information
     /// To get the full information, call 'CheckFileInProject' method on the result</para>
-    /// <para>All files except the one being checked are read from the FileSystem API</para>
     /// </summary>
     ///
     /// <param name="filename">The filename for the file.</param>
     /// <param name="source">The full source for the file.</param>
-    /// <param name="options">The options for the project or script, used to determine active --define conditionals and other options relevant to parsing.</param>
-    /// <param name="userOpName">An optional string used for tracing compiler operations associated with this request.</param>
-    member ParseFileInProject : filename: string * source: string * options: FSharpProjectOptions * ?userOpName: string -> Async<FSharpParseFileResults>
+    /// <param name="options">Parsing options for the project or script.</param>
+    member ParseFile: filename: string * source: string * options: FSharpParsingOptions -> Async<FSharpParseFileResults>
```

Comment:
Please add a ``ParseFileInProject`` for compatibility which extracts the parsing options from the project options.

Reply:
Should I mark it as deprecated?

Reply:
Should it have the same signature and include userOpName?

Reply:
@auduchinok I think don't mark it as deprecated yet - we would need to update all docs.  Add an item on https://github.com/fsharp/FSharp.Compiler.Service/issues to track this

Yes, please use the same signature , and yes include userOpName - we should include that for ``ParseFile`` even if we don't use it as we should pass in causality-tracking information for any operation that consumes CPU thread or Reactor-thread resources